### PR TITLE
[BREAKING] Promisify entire package

### PR DESCRIPTION
**Pull request by mvolz**
* Default export function returns a promise
* Most methods now syncronous instead of using callbacks
* Update version indicating non-backwards compatible
chanes
* Move test in index.js to test/index.js
* Move test_files to test/static

Bug: T96363

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,99 +5,47 @@
...
 
 // Default exported function
-exports = module.exports = function(urlOrOpts, callback){
-	request(urlOrOpts, function(error, response, html){
-		var chtml = cheerio.load(html);
-		exports.parseAll(chtml, function(err, results){
-			callback(err, results);
-		});
+exports = module.exports = function(urlOrOpts) {
+	return preq.get(urlOrOpts
```
Comment (d00rman) [inline index.js]: Note that, unlike `request`, `preq` rejects the promise if its status is >= 400, but we shouldn't handle it here, as this is a library.

File: index.js [hunk 2]
```diff
@@ -5,99 +5,47 @@
...
-		allMetadata = {},
-		metadataFunctions = exports.metadataFunctions;
+exports.parseAll = function(chtml){
+	var fcn;
+	var meta;
+	var allMetadata = {};
+	var metadataFunctions = exports.metadataFunctions;
 
-	async.forEach(Object.keys(metadataFunctions), function (key, cb){
+	Object.keys(metadataFunctions).forEach(function(key) {
```
Comment (d00rman) [inline index.js]: This processing can take quite some time. I'm wondering if we should split it in different Promise cycles.

File: index.js (hunk 1 above)
Comment (mvolz) [inline index.js]: Yeah, we don't actually use this function upstream in citoid- it's more of a convenience method to demo the library.

File: index.js (hunk 2 above)
Comment (mvolz) [inline index.js]: Working on it in new WIP commit. (Just letting you know because in IRC you mentioned that you might do this :))

**Comment by mvolz** [comment]
See branch promisifymerged

**Pull request closed without merging**